src/common/utxobased/engine/UtxoEngineProcessor.ts

```diff
@@ -950,165 +947,132 @@ export const pickNextTask = async (
         .catch(err => {
           addressTransactionCache[address] = state
           console.error(err)
-          args.log('error in updateTransactionCache:', err)
+          common.log('error in updateTransactionCache:', err)
         })
       return wsTask
     }
   }
 }
 
-interface UpdateTransactionsSpecificArgs extends CommonArgs {
-  serverUri: string
-  txId: string
-}
-
 const updateTransactionsSpecific = (
-  args: UpdateTransactionsSpecificArgs
+  common: CommonParams,
+  args: { serverUri: string; txId: string }
 ): WsTask<unknown> => {
-  const {
-    walletInfo,
-    emitter,
-    walletTools,
-    pluginInfo,
-    dataLayer,
-    serverUri,
-    txId,
-    taskCache
-  } = args
+  const { serverUri, txId } = args
   const deferred = new Deferred<unknown>()
   deferred.promise
     .then(async (txResponse: unknown) => {
       // Grab tx to update it
-      const txs = await dataLayer.fetchTransactions({ txId })
+      const txs = await common.dataLayer.fetchTransactions({ txId })
       let tx = txs[0]
       if (tx == null) return
 
-      if (pluginInfo.engineInfo.txSpecificHandling != null) {
+      if (common.pluginInfo.engineInfo.txSpecificHandling != null) {
         // Do coin-specific things to it
-        tx = pluginInfo.engineInfo.txSpecificHandling(tx, txResponse)
+        tx = common.pluginInfo.engineInfo.txSpecificHandling(tx, txResponse)
       }
 
       // Process and save new tx
-      const processedTx = await dataLayer.saveTransaction({
+      const processedTx = await common.dataLayer.saveTransaction({
         tx
       })
 
       await transactionChanged({
-        walletId: walletInfo.id,
-        emitter,
-        walletTools,
-        dataLayer: dataLayer,
-        pluginInfo,
+        walletId: common.walletInfo.id,
+        emitter: common.emitter,
+        walletTools: common.walletTools,
+        dataLayer: common.dataLayer,
+        pluginInfo: common.pluginInfo,
         tx: processedTx
       })
     })
     .catch(err => {
       console.error(err)
-      args.log('error in updateTransactionsSpecific:', err)
-      taskCache.updateTransactionSpecificCache[txId] = { processing: false }
+      common.log('error in updateTransactionsSpecific:', err)
+      common.taskCache.updateTransactionSpecificCache[txId] = {
+        processing: false
+      }
     })
 
-  return args.serverStates.transactionSpecialQueryTask(
+  return common.serverStates.transactionSpecialQueryTask(
     serverUri,
     txId,
     deferred
   )
 }
 
-interface UpdateTransactionsArgs extends CommonArgs {
-  needsTxSpecific: boolean
-  serverUri: string
-  txId: string
-}
-
 const updateTransactions = (
-  args: UpdateTransactionsArgs
+  common: CommonParams,
+  needsTxSpecific: boolean,
+  serverUri: string,
+  txId: string
```

Comment:
So here we have positional parameters, but above `updateTransactionsSpecific` has named parameters. I'm not sure if we want to be more consistent about named / positional parameters, but I do appreciate that `CommonParams` is cleaned up now.

This is just an observation. There are no changes needed unless you want to.